src/components/Onboarding/OnboardingChecklist/index.tsx

```diff
@@ -0,0 +1,158 @@
+    const didNotFinishOnboarding = !!checklistItems.find(({ group }) => {
+      const completedSteps = completedGroups[group]?.completedSteps;
+      return completedSteps?.length !== allSteps[group].length;
+    });
+
+    if (
+      checklistDismissals >= 3 ||
+      isChecklistVisible ||
+      !lastChecklistDismissal ||
+      !didNotFinishOnboarding
+    ) {
+      return;
+    }
+
+    const now = Date.now();
+    const oneMonth = 1000 * 60 * 60 * 24 * 30;
+    const lastDismissal = new Date(lastChecklistDismissal);
+
+    if (now - lastDismissal.getTime() > oneMonth) {
+      dispatch(setIsChecklistVisible(true));
+      logEvent('onboarding_checklist_auto_open', {
+        totalDismissals: checklistDismissals,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [completedGroups, checklistDismissals, lastChecklistDismissal]);
+
+  // If the onboarding is active, don't show the checklist
+  if (isActive) {
+    return null;
+  }
+
+  if (!isChecklistVisible) {
+    if (router.pathname === '/') {
+      return (
+        <Button
+          shape={ButtonShape.Circle}
+          className={classNames(styles.checklistPosition)}
+          tooltip={t('onboarding:onboarding-checklist')}
+          onClick={openChecklist}
+          size={ButtonSize.Large}
+        >
+          <IconQuestionMark className={styles.helpIcon} />
+        </Button>
+      );
+    }
+
+    return null;
+  }
+
+  const handleChecklistItemClick = async (group: OnboardingGroup) => {
+    const item = checklistItems.find((i) => i.group === group);
+    if (!item) return;
+
+    if (item.href) {
+      await router.push(item.href);
+      startTour(group);
+    }
+
+    logButtonClick('onboarding_checklist_item', {
+      group,
+    });
+  };
+
+  return (
+    <div className={classNames(styles.checklist, styles.checklistPosition)}>
+      <div className={styles.checklistHeader}>
+        <h4>{t('onboarding:onboarding-checklist')}</h4>
+
+        <Button
+          tooltip={t('close')}
+          shape={ButtonShape.Circle}
+          variant={ButtonVariant.Ghost}
+          ariaLabel={t('close')}
+          size={ButtonSize.Small}
+          onClick={handleDismiss}
+        >
+          <IconClose />
+        </Button>
+      </div>
+
+      <ul>
+        {checklistItems.map((item) => {
+          const checked =
+            completedGroups[item.group]?.completedSteps?.length === allSteps[item.group].length;
+
+          return (
+            <li key={item.group}>
+              <button
+                onClick={() => handleChecklistItemClick(item.group)}
+                type="button"
+                className={classNames(checked && styles.completed)}
+              >
+                <div className={classNames(styles.checkCircle, checked && styles.filled)}>
+                  {checked ? <CheckIcon /> : null}
+                </div>
+
+                <p>{item.title}</p>
```

Comment:
We should add a styling here to make sure the text sticks to the left/right depending on RTL or LTR

<img width="306" alt="Screenshot 2024-01-26 at 10 01 50 am" src="https://github.com/quran/quran.com-frontend-next/assets/15169499/f97bab81-9bdb-4c56-8a61-5850867d7e05">